packages/ckeditor5-coremedia-link/src/linktarget/config/LinkTargetConfig.ts

```diff
@@ -45,11 +58,70 @@ import { EditorConfig } from "@ckeditor/ckeditor5-core/src/editor/editorconfig";
  * this plugin, while `"myCustomTarget"` provides a custom fixed target.
  * `"myCustomTarget"` will be represented in model by `linkTarget="myCustomTarget"`
  * and will get a toggle button with given icon and title.
+ *
+ * It is also possible to provide a `filter` instead of a `type`
+ * for the default target rules.
  */
 interface LinkTargetConfig {
   targets?: (DefaultTarget | LinkTargetOptionDefinition)[];
+  defaultTargets?: TargetDefaultRuleDefinition[];
 }
 
+/**
+ * Parses a possibly existing configuration option as part of CKEditor's
+ * link plugin configuration. It expects an entry `defaultTargets` which contains an
+ * array of default targets for different kind of link types.
+ *
+ * @param config - CKEditor configuration to parse
+ */
+const parseDefaultLinkTargetConfig = (config: Config<EditorConfig>): TargetDefaultRuleDefinitionWithFilter[] => {
+  const fromConfig: unknown = config.get("link.defaultTargets");
+  const result: TargetDefaultRuleDefinitionWithFilter[] = [];
+
+  if (!Array.isArray(fromConfig)) {
+    throw new Error(
+      `link.defaultTargets: Unexpected configuration. Array expected but is: ${JSON.stringify(fromConfig)}`
+    );
+  }
+  const defaultTargetsArray: unknown[] = fromConfig;
+  defaultTargetsArray.forEach((entry: unknown): void => {
+    if (isTargetDefaultRuleDefinitionWithFilter(entry)) {
+      result.push(entry);
+    } else if (isTargetDefaultRuleDefinitionWithType(entry)) {
+      const externalLinkFilter = (url: string) => (url ? url.startsWith("https://") : false);
```

Comment:
A method which creates the filter would be great to reduce the loc in this function.

The function might grow if we add additional configuration.